### DISCUSSION TITLE
Help: LDAP configuration

### DISCUSSION THREAD

**Discussion by g0lgs** [nilsteampassnet/TeamPass#3842] [General]
Hi,

I am trying to fathom out how to get LDAP to work with Teampass 3.x but I am lost and not matter how much searching I do I am getting nowhere.

Is there some up-to-date documentation that might help ?

My Teampass is installed on a Ubuntu 22.04 Virtual Machine on the same VLAN as the Windows AD.

I have configured the LDAP settings to get to my Windows LDAP and the  'Test current configuration' at the bottom apparently suceeds, but I cannot fathom out how I'm supposed to add the user account into TeamPass or actually login using the account 

If I try to login with any LDAP account it just stays on the Login page with 'In Progress'  message at the top of the page.

**Comment by ismaelvillalba** [comment]
Hello @g0lgs! Have you explored the 'Users' section? You'll find an 'LDAP synchronization' button there. Could you share how you configured the LDAP settings to ensure the successful configuration test?

**Reply by g0lgs** [reply]
Hi,

I had tried the LDAP Synchronizatiom' that but it was not showing any accounts, however having had more time to try different settings I found that my attempts at setting a LDAP group object filter were apparently causing the system to be unable to find any accounts having removed that option it now finds accounts.

I have been unable to get a filter option working that restricts the list of available accounts to the group that I would like (i.e A 'Teampass_users' group in a given OU). It always shows me all accounts in my defined 'Additional User DN' (OU=xxxx_accounts) or it finds non at all (at present this is not much of an issue as the number of accounts in that OU is quite small - i.e. under 100).

**Reply by ismaelvillalba** [reply]
Hello again! Can you please provide me with the details of the keys you set up within the LDAP section? I've successfully tested the authentication, but I'm facing an issue when trying to synchronize the users. In the "Additional User DN" field, I haven't entered anything. Another unusual problem I've encountered, which I'm not sure if you've experienced as well, is that the "Enable users authentication through LDAP server" option occasionally disables itself every few minutes. Could you offer a solution for both this issue and the necessary enabling options?

**Comment by g0lgs** [comment]
I'm not sure what you mean by 'keys' ?

I have Hosts set to point at both my Windows DC's (i.e: 'dc01.xxx.local,dc02.xxx.local') , LDAP Port set to 389, Base DN to 'DC=xxx,DC=local', Username is set to 'CN=binder,DC=xxx,DC=local', User DN to 'distinguishedname', User Name Attribute to 'samaccountname', LDAP SSL and LDAP TLS are both off, Certificate check for LDAP TLS set to LDAP_OPT_X_TLS_NEVER, LDPAD GUID attribute set to 'objectguid'. Local and LDAP users = Yes.

I use the 'binder' account which is in the top level 'xxx.local'  OU for my GitLab LDAP intergration too.

**Comment by mercurysr** [comment]
Did you get it? I'm having the same problem.
